backend/app/routers/idioms.py

```diff
@@ -0,0 +1,133 @@
+from typing import Annotated
+from uuid import UUID
+
+from fastapi import APIRouter, Depends, HTTPException, Query
+from sqlalchemy import func
+from sqlalchemy import text as sql_text
+from sqlalchemy.orm import Session
+
+from app import database
+from app.models.idioms import IdiomModel
+from app.schemas.idioms import IdiomCreate, IdiomSchema, IdiomUpdate
+
+SessionDep = Annotated[Session, Depends(database.get_session)]
+
+router = APIRouter(prefix="/idioms", tags=["idioms"])
+
+
+@router.get("/", response_model=list[IdiomSchema])
+async def get_idioms(
+    db: SessionDep,
+    page: int = 1,
+    limit: Annotated[int, Query(le=50)] = 50,
+    text: Annotated[str, Query()] = "",
+    sort: Annotated[str | None, Query()] = None,
+) -> list[IdiomSchema]:
+    text = text.strip()
+    offset = (page - 1) * limit
+
+    query = db.query(IdiomModel)
+
+    if text:
+        query = query.filter(IdiomModel.text.ilike(f"%{text}%"))
+
+    match sort:
+        case "frequency":
+            query = query.order_by(IdiomModel.frequency_of_use.asc())
+        case "-frequency":
+            query = query.order_by(IdiomModel.frequency_of_use.desc())
+        case "imagery":
+            query = query.order_by(IdiomModel.literal_transparency.asc())
+        case "-imagery":
+            query = query.order_by(IdiomModel.literal_transparency.desc())
+        case _:
+            query = query.order_by(IdiomModel.text.asc())
+
+    return [
+        IdiomSchema.model_validate(idiom)
+        for idiom in (query.limit(limit).offset(offset).all())
+    ]
+
+
+@router.get("/random", response_model=list[IdiomSchema])
+async def get_random_idioms(
+    db: SessionDep,
+    page: int = 1,
+    limit: Annotated[int, Query(le=50)] = 50,
+    seed: Annotated[int | None, Query()] = None,
+) -> list[IdiomSchema]:
```

Comment:
Allows to fetch a random subset of idioms. The seed parameter allows to keep the random order stable so that pagination does not repeat the same idioms in previous pages.